test/jdk/javax/swing/plaf/basic/BasicPasswordFieldUI/PasswordSelectionWordTest.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test
+ * @key headful
+ * @bug 4231444 8354646
+ * @summary Password fields' ActionMap needs to replace
+ * DefaultEditorKit.selectWordAction with DefaultEditorKit.selectLineAction.
+ * When we failed to do this: the user could double-click words and
+ * identify spaces in the password. 8354646 is an offshoot of 4231444 that
+ * reproduced in Aqua because AquaTextPasswordFieldUI does NOT extend
+ * BasicPasswordFieldUI.
+ *
+ * @run main PasswordSelectionWordTest
+ */
+
+import javax.swing.Action;
+import javax.swing.JPasswordField;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+import javax.swing.UnsupportedLookAndFeelException;
+import javax.swing.plaf.basic.BasicTextUI;
+import javax.swing.text.DefaultEditorKit;
+import java.awt.event.ActionEvent;
+
+public class PasswordSelectionWordTest {
+    public static void main(String[] args) throws Exception {
+        for (UIManager.LookAndFeelInfo laf :
+                UIManager.getInstalledLookAndFeels()) {
+            System.out.println("Testing LAF: " + laf.getClassName());
+            SwingUtilities.invokeAndWait(() -> setLookAndFeel(laf));
```

Comment:
If setLookAndFeel finds UnsupportedLookAndFeelException then the runTest here is still executed here whereas it shouldn't continue for that L&F...guess you can use a boolean return value for setLookAndFeel.

Reply:
This is updated. (I have no preference on the behavior; this was copied and pasted from other test files.)

Reply:
I guess I am for not treating UnsupportedLookAndFeelException as an error. It will cause  the test to stop executing the next L&F if there is one in the list after the unsupported L&F. 

I have seen this exception in CI testing for GTK L&F but I guess it's for headless environment so I agree it's rare and will not happen in normal case..I was just thinking why to cause unnecessary execution..It maynot matter much for this test but for tests which uses Robot and has delay/waitForIdle executing already run L&F because of UnsupportedException will cause additional cycle.